### PR TITLE
Add two swift feature and make a full translation for swift to zh-cn

### PR DESCRIPTION
**Pull request by kamidox**
### Add two swift feature:

* Optional init
* Enum with associated values

### Make full translation for swift to zh-cn

### CODE REVIEW

File: zh-cn/swift-cn.html.markdown [hunk 1]
```diff
@@ -5,223 +5,582 @@ contributors:
...
-let label = "some text " + String(myVariable)     // Casting
-let piText = "Pi = \(myConstant)"                 // String interpolation
-var optionalString: String? = "optional"          // Can be nil
-optionalString = nil
+let explicitDouble: Double = 70   // 明确指定变量类型为 Double ，否则编译器将自动推断变量类型
+let weak = "keyword"; let override = "another keyword" // 语句之间可以用分号隔开，语句未尾不需要分号
+let intValue = 0007 // 7
+let largeIntValue = 77_000 // 77000
+let label = "some text " + String(myVariable) // 类型转换
+let piText = "Pi = \(π), Pi 2 = \(π * 2)" // 格式化字符串
```
Comment (geoffliu) [inline zh-cn/swift-cn.html.markdown]: Shouldn't `π` be defined first?

File: zh-cn/swift-cn.html.markdown [hunk 2]
```diff
@@ -5,223 +5,582 @@ contributors:
...
+            return 4 * sideLength
+        }
+        set {
+            // `newValue` 是个隐含的变量，它代表设置进来的值
+            sideLength = newValue / 4
+        }
+    }
+    
+    // 延时加载的属性，只有这个属性第一次被引用时才进行加载，而不是定义时就加载
+    // subShape值为 nil ，走到 suShape 第一次被引用时才初始化
```
Comment (geoffliu) [inline zh-cn/swift-cn.html.markdown]: Typo

File: zh-cn/swift-cn.html.markdown [hunk 3]
```diff
@@ -5,223 +5,582 @@ contributors:
...
+// MARK: 协议
+// 与 Java 的 interface 类似
 //
 
-// `协议(protocol)`: 与Java 的接口(Interface) 类似.
-// `扩展(extension)`: 为现有类型添加额外特性
-// 泛型: 与Java 相似。使用`where` 关键字指定
-//   泛型的要求.
+// 协议可以让遵循同一协议的类型实例拥有相同的属性，方法，类方法，操作符或下标运算符
+// 定义一个协议，这个协议饮食 enabled 计算属性且包含 buildShape 方法
```
Comment (geoffliu) [inline zh-cn/swift-cn.html.markdown]: 饮食？

**Comment by geoffliu** [comment]
Thanks for the work, just a few minor comments, then this is good to go.

**Comment by kamidox** [comment]
Sorry for the typo, I reviewed all of them this time.

File: zh-cn/swift-cn.html.markdown [hunk 4]
```diff
@@ -5,223 +5,587 @@ contributors:
...
+// 定义一个前缀运算符，使矩形的边长放大三位
+prefix func !!! (inout shape: Square) -> Square {
+    shape.sideLength *= 3
+    return shape
+}
+
+// 当前值
+println(mySquare.sideLength) // 4
+
+// 使用自定义的 !!! 运算符来把矩形边长放大三位
```
Comment (geoffliu) [inline zh-cn/swift-cn.html.markdown]: 三倍？

File: zh-cn/swift-cn.html.markdown [hunk 5]
```diff
@@ -5,223 +5,587 @@ contributors:
...
+let foundAtIndex = findIndex([1, 2, 3, 4], 3)
+println(foundAtIndex == 2) // true
+
+// 自定义运算符:
+// 自定义运算符可以以下面的字符打头:
+//      / = - + * % < > ! & | ^ . ~
+// 甚至是 Unicode 的数学运算符等
+prefix operator !!! {}
+
+// 定义一个前缀运算符，使矩形的边长放大三位
```
Comment (geoffliu) [inline zh-cn/swift-cn.html.markdown]: 三倍？

**Comment by geoffliu** [comment]
One last thing. :)

**Comment by kamidox** [comment]
Still one fugitive. :)

File: zh-cn/swift-cn.html.markdown [hunk 6]
```diff
@@ -5,223 +5,587 @@ contributors:
...
+let foundAtIndex = findIndex([1, 2, 3, 4], 3)
+println(foundAtIndex == 2) // true
+
+// 自定义运算符:
+// 自定义运算符可以以下面的字符打头:
+//      / = - + * % < > ! & | ^ . ~
+// 甚至是 Unicode 的数学运算符等
+prefix operator !!! {}
+
+// 定义一个前缀运算符，使矩形的边长放大倍
```
Comment (geoffliu) [inline zh-cn/swift-cn.html.markdown]: Still not quite there. :)

File: zh-cn/swift-cn.html.markdown (hunk 4 above)
Comment (geoffliu) [inline zh-cn/swift-cn.html.markdown]: And this one

**Comment by kamidox** [comment]
Hope it's there this time.

**Comment by geoffliu** [comment]
Ha thanks! Sorry for being so nit-picky.

**Comment by kamidox** [comment]
Thanks your time for review. :)

**Pull request merged**